test-gen-plugin/src/main/java/com/dylibso/chicory/maven/JavaTestGen.java

```diff
@@ -151,27 +144,39 @@ public CompilationUnit generate(
                     }
                     String hostFuncs =
                             detectImports(importsName, lastModuleVarName, importsSourceRoot);
-                    testClass.addFieldWithInitializer(
-                            parseClassOrInterfaceType("TestModule"),
-                            lastModuleVarName,
-                            generateModuleInstantiation(
-                                    cmd, currentWasmFile, importsName, hostFuncs));
                     testClass.addFieldWithInitializer(
                             "Instance",
                             lastInstanceVarName,
-                            new NameExpr(lastModuleVarName + ".instance()"));
+                            new NullLiteralExpr(),
+                            Modifier.Keyword.PUBLIC,
+                            Modifier.Keyword.STATIC);
+
+                    var instantiateMethodName = "instantiate_" + lastInstanceVarName;
+                    var instantiateMethod =
+                            testClass.addMethod(instantiateMethodName, Modifier.Keyword.PUBLIC);
+                    // It needs to be a test to be executed
+                    instantiateMethod.addAnnotation("Test");
```

Comment:
this artificially increases the number of tests we are executing, but I have no better way to keep the assertions separated and in order.